pkg/parquetquery/iters.go

```diff
@@ -1165,17 +1212,31 @@ func (c *SyncIterator) next() (RowNumber, *pq.Value, error) {
 		for c.currBufN < len(c.currBuf) {
 			v := &c.currBuf[c.currBufN]
 
+			// If the repetition level is set, and the repetition level of the value is less or equal to
+			// the iterators repetition level, the v is either a single value or the start of a new array.
+			if c.repLevel < 0 || (v.RepetitionLevel() <= c.repLevel) {
+				if matchEndOfArray {
+					// If we are looking for the end of the array, we've now consumed all values and can return.
+					return c.curr, nil
+				}
+				// Otherwise we have to reset the current values buffer.
+				c.currValues = c.currValues[:0]
+			}
+
 			// Inspect all values to track the current row number,
 			// even if the value is filtered out next.
 			c.curr.Next(v.RepetitionLevel(), v.DefinitionLevel())
 			c.currBufN++
 			c.currPageN++
 
-			if c.filter != nil && !c.filter.KeepValue(*v) {
+			c.currValues = append(c.currValues, *v)
```

Comment:
i'm concerned about the perf impact of repeatedly appending to a slice and throwing it away. i know it's not allocing but this line will still get hit millions of times in a traceql query.

i suppose benches would show anything concerning.

Reply:
this is not showing up in benches, and then I looked at the benchmark diff to see if this shows up.

I took some CPU and Memprofiles from benchmarks, and diff'ed them, and It's coming under 1% (Cumulative) in the profile diffs.

here is how did the benchmark diffs:
- `go tool pprof -http :9999 -normalize -diff_base=<profile on main> <profile from this branch>` for CPU and Mem profiles
- search for  and `next` calls on `SyncIterator` Top table